### PR TITLE
add packer-plugin-docker package

### PR DESCRIPTION
**Pull request by kranurag7**
no tests given we don't have packer

### CODE REVIEW

File: packer-plugin-docker.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    with:
+      deps: github.com/go-jose/go-jose/v3@v3.0.3 github.com/hashicorp/go-retryablehttp@v0.7.7 golang.org/x/crypto@v0.17.0 golang.org/x/net@v0.23.0 google.golang.org/grpc@v1.56.3 google.golang.org/protobuf@v1.33.0 github.com/hashicorp/packer-plugin-sdk@v0.5.4
+      replaces: github.com/go-jose/go-jose/v3=github.com/go-jose/go-jose/v3@v3.0.3 gopkg.in/square/go-jose.v2=github.com/go-jose/go-jose/v3@v3.0.3
+
+  - uses: go/build
+    with:
+      packages: .
+      output: packer-plugin-docker
+      ldflags: -X github.com/hashicorp/packer-plugin-docker/version.Version=${{package.version}}
+
```
Comment (mamccorm) [inline packer-plugin-docker.yaml:25]: Does the binary respond to a --version or --help? Given it's a go binary it's highly likely, and we could at least add a basic test

Comment (kranurag7) [reply packer-plugin-docker.yaml:25]: Thanks Mark, I've added the tests along with notes via https://github.com/kranurag7/os/commit/2aac0c27cc958983b36867cbad01a8053b5b9b92

Comment (mamccorm) [reply packer-plugin-docker.yaml:25]: Thanks for following up. the absence of a --help or a --version, can you share info on how we were able to manually test this? It seems like it requires packer to operate, and I don't believe we have a packer apk today

Comment (mamccorm) [reply packer-plugin-docker.yaml:25]: Have we been able to do any manual validation that the plugins are detected and usable by packer?

**Comment by kranurag7** [comment]
closing this for now given this requires more work and we don't have confirmation yet on the tag.

**Pull request closed without merging**